File: 1.4d/1_4dFunctions.cpp
#include <iostream>
 
void doIt(int x)
{
    int y = 4;
    std::cout << "doIt: x = " << x << " y = " << y << std::endl;
    x = 3;
    std::cout << "doIt: x = " << x << " y = " << y << std::endl;
}
 
int main()
{
    int x = 1;
    int y = 2;
    std::cout << "main: x = " << x << " y = " << y << std::endl;
    doIt(x);
    std::cout << "main: x = " << x << " y = " << y << std::endl;
    return 0;
}

File: 1.4d/1_4dFunctions.ts
function doIt(x: number){
    let y: number = 4;
    console.log(`doIt: x= ${x}, y= ${y}`);
    x = 3;
    console.log(`doIt: x= ${x}, y= ${y}`);
}

(function main(){
  let x: number = 1;
  let y: number = 2;
  
  console.log(`main: x= ${x}, y= ${y};`)
  doIt(x);
  console.log(`main: x= ${x}, y= ${y};`)
})();